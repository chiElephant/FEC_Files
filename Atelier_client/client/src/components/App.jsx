import React, { useState, useEffect } from 'react';
import ReviewsModule from './reviews_module/ReviewsModule.jsx'

function App(props) {
  return (
    <div>
      <h1> The RedBean Atelier App </h1>
      <ReviewsModule
        productId={71698}
        productName={'Bright Future Sunglasses'}/>
    </div>
  );
}

export default App;
