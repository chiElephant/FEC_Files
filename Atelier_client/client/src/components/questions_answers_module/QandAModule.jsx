import React, { useState, useEffect } from 'react';
import SearchQ from './SearchQ.jsx';
import QuestionList from './QuestionList.jsx';
import getQuestions from './helper_functions/getQuestions.js';

function QandAModule({ product_id, product_name }) {
  // set props at {product_id, product_name}
  const [questionList, setQuestionList] = useState([]);
  const [page, setPage] = useState(1);
  const [count, setCount] = useState(100);

  useEffect(() => {
    getQuestions(product_id, page, count, setQuestionList);
  }, []);

  return (
    <div id="QandAtop">
      <h1>Questions & Answers</h1>
      <SearchQ />
      <QuestionList
        questionList={questionList}
        productName={product_name}
        productId={product_id}
      />
    </div>
  );
}

export default QandAModule;
