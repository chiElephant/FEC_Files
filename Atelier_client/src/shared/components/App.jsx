import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Topbar from './Topbar.jsx';
import routes from '../routes';

export default function App({ serverData = null }) {
  return (
    <>
      <Topbar />
      <Routes>
        {routes.map((route) => {
          const { path, component: Component } = route;
          return (
            <Route
              key={path}
              path={path}
              element={<Component data={serverData} />}
            />
          );
        })}
      </Routes>
    </>
  );
}
