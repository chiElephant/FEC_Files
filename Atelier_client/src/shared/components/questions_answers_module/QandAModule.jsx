import React, { useState, useEffect } from 'react';
import SearchQ from './SearchQ.jsx';
import QuestionList from './QuestionList.jsx';
import useGetQuestions from './custom_hooks/useGetQuestions.jsx';
import useDebounce from './custom_hooks/useDebounce.jsx';
import useFilterByMatchingText from './custom_hooks/useFilterByMatchingText.jsx';

function QandAModule({ questions_answers, product_id, product_name }) {
  const [questionsWithAnswersData, setQuestionsWithAnswersData] =
    useState(questions_answers);
  const [displayList, setDisplayList] = useState([]);
  const [countShown, setCountShown] = useState(2);
  const [showMoreQuestions, setShowMoreQuestions] = useState(false);
  const [fetchQuestions, setFetchQuestions] = useState(false);

  useEffect(() => {
    setDisplayList(questionsWithAnswersData.slice(0, countShown));

    if (countShown < questionsWithAnswersData.length) {
      setShowMoreQuestions(true);
    } else {
      setShowMoreQuestions(false);
    }
  }, [questionsWithAnswersData, countShown]);

  //filtered questions state
  const [displayFiltered, setDisplayFiltered] = useState([]);
  const [filterCountShown, setFilterCountShown] = useState(2);
  const [showMoreFilteredQuestions, setShowMoreFilteredQuestions] =
    useState(false);

  //search text state
  const [searchText, setSearchText] = useState('');
  const [filterMode, setFilterMode] = useState(false);
  const debouncedSearchText = useDebounce(searchText, 1500);

  const filteredList = useFilterByMatchingText(
    displayList,
    searchText,
    'question_body'
  );

  //as we incement count, also increment questions to display by two
  // useEffect(() => {
  //   if (countShown < displayList.length) {
  //     let newList;
  //     if (countShown === 2) {
  //       newList = displayList;
  //     } else {
  //       const grabNextTwo = displayList.slice(countShown - 2, countShown);
  //       newList = [...displayList, ...grabNextTwo];
  //     }

  //     setDisplayList(newList);
  //     setShowMoreQuestions(true);
  //   }
  // }, [countShown]);

  //as we increment filtered count, also increment filtered questions to display by two
  // useEffect(() => {
  //   if (displayFiltered.length !== filterCountShown) {
  //     if (filterCountShown < filteredList.length) {
  //       let newList;
  //       if (filterCountShown === 2) {
  //         newList = displayFiltered;
  //       } else {
  //         const grabNextTwo = filteredList.slice(
  //           filterCountShown - 2,
  //           filterCountShown
  //         );
  //         newList = [...displayFiltered, ...grabNextTwo];
  //       }

  //       setDisplayFiltered(newList);
  //       setShowMoreFilteredQuestions(true);
  //     } else {
  //       setDisplayFiltered(filteredList);
  //       setShowMoreFilteredQuestions(false);
  //     }
  //   }
  // }, [filterCountShown]);

  // handle searching, want to only display first two questions
  // useEffect(() => {
  //   if (debouncedSearchText.length >= 3) {
  //     const grabFirstTwo = filteredList.slice(0, 2);

  //     if (filterCountShown < filteredList.length) {
  //       setShowMoreFilteredQuestions(true);
  //     } else {
  //       setShowMoreFilteredQuestions(false);
  //     }
  //     setFilterMode(true);
  //     setCountShown(2);
  //     setFilterCountShown(2);
  //     setDisplayFiltered(grabFirstTwo);
  //   } else {
  //     const grabFirstTwo = displayList.slice(0, 2);

  //     setFilterMode(false);
  //     setFilterCountShown(2);
  //     setDisplayList(grabFirstTwo);
  //   }
  // }, [debouncedSearchText]);

  const handleUpdateSearchText = (e) => {
    setSearchText(e.target.value);
  };

  const handleShowMoreQuestions = () => {
    console.log('countShown: ', countShown);
    console.log('displayList.length: ', displayList.length);
    console.log(countShown < displayList.length);

    setCountShown((prevState) => prevState + 2);
    console.log(countShown);
  };

  const handleShowMoreFilteredQuestions = () => {
    setFilterCountShown((prevState) => prevState + 2);
  };

  const handleFetchQuestions = () => {
    setFetchQuestions((prevState) => !prevState);
    setShowMoreQuestions(false);
  };

  const list = filterMode ? (
    <QuestionList
      handleFetchQuestions={handleFetchQuestions}
      displayList={displayFiltered}
      productName={product_name}
      productId={product_id}
      showMoreQuestions={showMoreFilteredQuestions}
      handleShowMoreQuestions={handleShowMoreFilteredQuestions}
    />
  ) : (
    <QuestionList
      handleFetchQuestions={handleFetchQuestions}
      displayList={displayList}
      productName={product_name}
      productId={product_id}
      showMoreQuestions={showMoreQuestions}
      handleShowMoreQuestions={handleShowMoreQuestions}
    />
  );

  return (
    <div id="QandA-main">
      <div id="QandAtop">
        <h1 id="qanda-header">Questions & Answers</h1>
        <SearchQ
          handleUpdateSearchText={handleUpdateSearchText}
          searchText={searchText}
        />
        {list}
      </div>
    </div>
  );
}

export default QandAModule;
