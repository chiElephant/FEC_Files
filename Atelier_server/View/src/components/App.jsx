import axios from 'axios';
import React, { useState, useEffect } from 'react';

import { useParams } from 'react-router-dom';
import Topbar from './Topbar.jsx';
import Overview from './overview_module/Overview.jsx';
// import QandAModule from './questions_answers_module/QandAModule.jsx';
// import ReviewsModule from './reviews_module/ReviewsModule.jsx';
// import ProductLinks from './ProductLinks.jsx';

function App() {
  const { id } = useParams();
  const [currentProduct, setCurrentProduct] = useState(null);
  // const [products, setProducts] = useState([]);

  useEffect(() => {
    const product_id = id || 1;
    axios.get(`/products/${product_id}`).then((res) => {
      setCurrentProduct(res.data);
    });
  }, [id]);

  // useEffect(() => {
  //   const options = {
  //     method: 'get',
  //     url: `/products`,
  //   };

  //   axios(options)
  //     .then((response) => {
  //       const productList = response.data;
  //       setProducts([...productList]);
  //     })
  //     .catch((error) => {
  //       console.log(error);
  //     });
  // }, []);

  const widgets =
    currentProduct === null ? (
      ''
    ) : (
      <>
        <Overview product_id={currentProduct.id} product={currentProduct} />
        {/* <QandAModule product_id={productId} product_name={productName} />
        <ReviewsModule product_id={productId} product_name={productName} /> */}
      </>
    );

  return (
    <div>
      <Topbar />
      {widgets}
      {/* <ProductLinks products={products} /> */}
    </div>
  );
}

export default App;
