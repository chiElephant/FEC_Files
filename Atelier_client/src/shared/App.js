/* global Image */

import React, { useState, useEffect, Suspense } from 'react';
import { useParams } from 'react-router-dom';

import axios from 'axios';
import Loading from './components/Loading.jsx';
import QandAModule from './components/questions_answers_module/QandAModule.jsx';
import ReviewsModule from './components/reviews_module/ReviewsModule.jsx';
import RelatedProducts from './components/RelatedProducts.jsx';
import Overview from './components/overview_module/Overview.jsx';

function App() {
  const { id: paramId } = useParams();
  const [productData, setProductData] = useState(null);
  const [relatedProducts, setRelatedProducts] = useState({});
  // const [loading, setLoading] = useState(true);
  // const [error, setError] = useState(null);

  console.log('productData: ', productData);
  console.log('paramId: ', paramId);

  // if (loading) {
  //   return <Loading />;
  // }

  // if (error) {
  //   return <div>Error: {error.message}</div>;
  // }

  return (
    <div id="components">
      <Overview productData={productData} />
      <RelatedProducts relatedProducts={relatedProducts} />
      <QandAModule
        product_id={productData.id}
        product_name={productData.name}
        questions_answers={productData.questionsWithAnswers}
      />
      <ReviewsModule
        product_id={productData.id}
        product_name={productData.name}
        reviewsData={productData.reviews}
        metaData={productData.metaData}
      />
    </div>
  );
}

export default App;
