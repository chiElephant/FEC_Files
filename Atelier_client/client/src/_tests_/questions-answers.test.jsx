/* global test, expect */

import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../components/App';
import QandAModule from '../components/questions_answers_module/QandAModule.jsx';

test('renders the landing page', () => {
  render(<App />);
});

// QandAModule Module
// test('Questions & Answers module renders correctly', () => {
//   const tree = renderer.create(<QandAModule />).toJSON();
//   expect(tree).toMatchSnapshot();
// });
