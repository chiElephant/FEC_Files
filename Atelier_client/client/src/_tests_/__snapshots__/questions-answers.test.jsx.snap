// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions & Answers module renders correctly 1`] = `
<div>
  <h1>
    Questions & Answers
  </h1>
</div>
`;

exports[`Questions & Answwers module renders correctly 1`] = `
<div>
  <h1>
    Questions & Answers
  </h1>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div>
    <div>
      <button>
         More Answered Questions 
      </button>
      <button
        onClick={[Function]}
      >
         Add Question + 
      </button>
    </div>
  </div>
</div>
`;
